### PR TITLE
SyntheticToken.sol natspec

### LINKED ISSUE #1051: NFCT NatSpec pre-audit
Some contracts within the NFCT conform to NatSpec. This should be updated such that all do for the OZ audit.

### PR DESCRIPTION
**Pull request by chrismaree**
last contract to close #1051

### CODE REVIEW

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 1]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
 
 
 /**
- * @notice A burnable and mintable ERC20. The contract deployer will initially
- * be the only minter and burner as well as the owner who is capable of adding new roles.
- * The contract deployer will also be the only initial minter of the contract.
+ * @title Burnable and mintable ERC20.
+ * @dev The contract deployer will initially be the only minter and burner as well as the owner  who
+ * is capable of adding new roles. The contract deployer will also be the only initial minter of the contract.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: nit: can remove the last sentence "The contract deployer will also be the only initial minter of the contract." which is repetitive

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 2]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
  */
+
 contract SyntheticToken is ExpandedERC20, ERC20Detailed {
+    /**
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Just checking, have you always made comments for constructors?

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 3]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     * @notice Constructs the SyntheticToken
+     * @param tokenName used to describe the new token.
+     * @param tokenSymbol short ticker abbreviation of the name. Ideally < 5 chars.
+     * @param tokenDecimals used to delimitate the new token.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as https://github.com/UMAprotocol/protocol/pull/1142#discussion_r400177693

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 4]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
     {}
 
     /**
-     * @dev Add minter role to account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Add minter role to account.
+     * @dev the caller must have the Owner role.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: If `Owner` role is capitalized, then should also make `Minter` role capitalized.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 5]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
 
     /**
-     * @dev Remove minter role from account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Remove minter role from account.
+     * @dev the caller must have the Owner role.
+     * @param account to be removed from the miner roll.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: role not roll

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 6]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
-        return holdsRole(uint(Roles.Minter), account);
-    }
-
     /**
-     * @dev Add burner role to account
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Add burner role to account
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above, `Burner`, `Minter`, `Owner` should all be capitalized or not for consistency

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 7]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     * @dev the caller must have the Owner role.
+     * @param account to be added as a burner roll.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: role not roll

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 8]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
         addMember(uint(Roles.Burner), account);
     }
 
     /**
-     * @dev Removes burner role from account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Removes burner role from account.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comments as above

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 9]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
-
     /**
-     * @dev Reset Owner role to account
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Reset Owner role to account
+     * @dev the caller must have the Owner role.
+     * @param account have it's rolls reset.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: this comment isn't very clear, perhaps would be better as `account will obtain the Owner role`

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 10]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
      */
     function resetOwner(address account) external {
         resetMember(uint(Roles.Owner), account);
     }
+
+    function isMinter(address account) public view returns (bool) {
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: +1 moving view functions to bottom

File: core/contracts/financial-templates/implementation/SyntheticToken.sol (hunk 2 above)
Comment (chrismaree) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: yup, always. Can strip them out of all the contracts if we want but to be consistent they have always been there

File: core/contracts/financial-templates/implementation/SyntheticToken.sol (hunk 4 above)
Comment (chrismaree) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: good point. fixed

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 11]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
 
     /**
-     * @dev Remove minter role from account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Remove Minter role from account.
+     * @dev the caller must have the Owner role.
+     * @param account to be removed from the miner role.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: typo: Minter

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 12]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
-
     /**
-     * @dev Reset Owner role to account
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Reset Owner role to account.
+     * @dev the caller must have the Owner role.
+     * @param account have it's roles reset.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: (I thought i commented on this but I don't see it anymore) Should make this clearer: `account the new holder of the Owner role`

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 13]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
- * The contract deployer will also be the only initial minter of the contract.
+ * @title Burnable and mintable ERC20.
+ * @dev The contract deployer will initially be the only minter and burner as well as the owner  who
+ * is capable of adding new roles.
  */
+
 contract SyntheticToken is ExpandedERC20, ERC20Detailed {
+    /**
+     * @notice Constructs the SyntheticToken.
+     * @param tokenName used to describe the new token.
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: nit: prefer to not use the name of the parameter as the beginning of the descriptive sentence.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 14]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     * @param tokenSymbol short ticker abbreviation of the name. Ideally < 5 chars.
+     * @param tokenDecimals used to define the precision used in the tokens numerical representation.
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 15]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
 
     /**
-     * @dev Add minter role to account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Add Minter role to account.
+     * @dev the caller must have the Owner role.
+     * @param account to be added to the Minter role.
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol (hunk 11 above)
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 16]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
-
     /**
-     * @dev Add burner role to account
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Add Burner role to account.
+     * @dev the caller must have the Owner role.
+     * @param account to be added as a Burner role.
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 17]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
 
     /**
-     * @dev Removes burner role from account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Removes Burner role from account.
+     * @dev the caller must have the Owner role.
+     * @param account to be removed from the Burner role.
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same comment as above.

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 18]
```diff
@@ -4,76 +4,73 @@ import "../../common/implementation/ExpandedERC20.sol";
...
-     * - the caller must have the Owner role.
+     * @notice Reset Owner role to account.
+     * @dev the caller must have the Owner role.
+     * @param account have it's roles reset.
      */
     function resetOwner(address account) external {
         resetMember(uint(Roles.Owner), account);
     }
+
+    function isMinter(address account) public view returns (bool) {
```
Comment (mrice32) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: No natspec on `isMinter` and `isBurner`?

File: core/contracts/financial-templates/implementation/SyntheticToken.sol (hunk 13 above)
Comment (nicholaspai) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: I think the linter actually requires you to put the name of the parameter after the `@param` tag

Comment (nicholaspai) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: The format I recall is `@param variable-name description...`

Comment (mrice32) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: Sorry, let me clarify. I think the name should be there, but it shouldn't be considered part of the sentence.
Prefer:
```
@param tokenName the percentage of blah to blah
```
over
```
@param tokenName is the percentage of blah to blah
```

Comment (mrice32) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: @ptare pointed this out a while back, but the latter is confusing when rendered on the docs site.

Comment (chrismaree) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: yup, thanks @mrice32 . I have refactored accordingly

File: core/contracts/financial-templates/implementation/SyntheticToken.sol (hunk 12 above)
Comment (chrismaree) [reply core/contracts/financial-templates/implementation/SyntheticToken.sol]: done

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 19]
```diff
@@ -4,76 +4,83 @@ import "../../common/implementation/ExpandedERC20.sol";
...
 
     /**
-     * @dev Removes burner role from account.
-     *
-     * Requirements:
-     *
-     * - the caller must have the Owner role.
+     * @notice Removes Burner role from account.
+     * @dev The caller must have the Owner role.
+     * @param account The address from which the Burner roll is removed.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: role not roll

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 20]
```diff
@@ -4,76 +4,83 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     * @param account The new holder of the Owner role.
      */
     function resetOwner(address account) external {
         resetMember(uint(Roles.Owner), account);
     }
+
+    /**
+     * @notice Checks if a given account holds the Minter role.
+     * @param account The address which is checked for the Minter role.
+     * @return bool True of False if the provided account is a Minter.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: possible typo: maybe change to "True if the provided account is a Minter"

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 21]
```diff
@@ -4,76 +4,83 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     */
+    function isMinter(address account) public view returns (bool) {
+        return holdsRole(uint(Roles.Minter), account);
+    }
+
+    /**
+     * @notice Checks if a given account holds the Burner role.
+     * @param account The address which is checked for the Burner role.
+     * @return bool True of False if the provided account is a Burner.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: Same

File: core/contracts/financial-templates/implementation/SyntheticToken.sol [hunk 22]
```diff
@@ -4,76 +4,83 @@ import "../../common/implementation/ExpandedERC20.sol";
...
+     * @return bool True if the provided account is a Minter.
+     */
+    function isMinter(address account) public view returns (bool) {
+        return holdsRole(uint(Roles.Minter), account);
+    }
+
+    /**
+     * @notice Checks if a given account holds the Burner role.
+     * @param account The address which is checked for the Burner role.
+     * @return bool True if the provided account is a Burner.
```
Comment (nicholaspai) [inline core/contracts/financial-templates/implementation/SyntheticToken.sol]: you a burner bro?

**Pull request merged**